internal/builder/builder.go

```diff
@@ -0,0 +1,279 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apiv1 "github.com/substratusai/substratus/api/v1"
+	"github.com/substratusai/substratus/internal/cloud"
+	"github.com/substratusai/substratus/internal/resources"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ptr "k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	saName = "container-builder"
+)
+
+type BuildableObject interface {
+	client.Object
+	GetContainer() apiv1.Container
```

Comment:
should GetContainer return a pointer such that you don't need another SetContainer?

Reply:
Updated it to do this